crates/turborepo-paths/src/anchored_system_path_buf.rs

```diff
@@ -115,6 +137,29 @@ impl AnchoredSystemPathBuf {
         self.0.as_str()
     }
 
+    pub fn new_unchecked(path: impl Into<Utf8PathBuf>) -> Self {
+        AnchoredSystemPathBuf(path.into())
+    }
+
+    // Takes in a path that has already been validated as anchored
+    // via `check_name` in `turborepo-cache` and constructs an
+    // `AnchoredSystemPathBuf` with no trailing slashes.
+    pub fn from_validated_tar_path(path: &str) -> Self {
+        let path = Utf8Path::new(path);
```

Comment:
How about a constructor that does the right thing, instead of `check_name`? This is essentially `new_unchecked`. 

`AnchoredSystemPathBuf::from_cleaned(raw) -> Result<(Self, bool), PathError>` to handle the extra windows-safe bit

Reply:
I wanna keep as little tar-specific code in turbopath as possible. Otherwise we have an awkward split of some cache code in turbopath and some in turborepo-cache